### PR TITLE
Add ability to skip steps

### PR DESCRIPTION
**Pull request by aeviiq**

### CODE REVIEW

**Comment by aeviiq** [comment]
Solves https://github.com/aeviiq/form-flow/issues/41

File: src/Transitioner.php [hunk 1]
```diff
@@ -110,8 +111,28 @@ public function forwards(FormFlowInterface $flow): Status
...
+                ++$increment;
+            } else {
+                $context->unsetHardSkipped($nextStep);
+            }
+
+            if ($skipEvent->isSoftSkipped()) {
+                $context->setSoftSkipped($nextStep);
+                ++$increment;
+            } else {
+                $context->unsetSoftSkipped($nextStep);
```
Comment (nvdbeek) [inline src/Transitioner.php]: Is the array always filled ?

File: src/Context.php [hunk 2]
```diff
@@ -76,4 +90,63 @@ public function isCompleted(StepInterface $step): bool
...
+    }
+
+    private function validateSkipRequest(StepInterface $step): void
+    {
+        $stepNumber = $step->getNumber();
+        if ($stepNumber < 1 || $stepNumber > $this->totalNumberOfSteps) {
+            throw new LogicException(sprintf('Step number "%d" is invalid for this context.', $stepNumber));
+        }
+
+        if ($stepNumber < 2) {
```
Comment (nvdbeek) [inline src/Context.php]: future improvements: setting the first step number to support more goals.

File: src/Transitioner.php (hunk 1 above)
Comment (aeviiq) [reply src/Transitioner.php]: no, but unset() is called directly inside the method.

**Pull request merged**